pkg/vault/contracts/Vault.sol

```diff
@@ -1160,11 +1160,25 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
         }
 
         if (params.kind == SwapKind.EXACT_IN) {
+            if (amountOutRaw < _MINIMUM_WRAP_AMOUNT) {
+                // If amount calculated is too small, rounding issues can be introduced that favor the user and can
+                // drain the buffer. _MINIMUM_WRAP_AMOUNT prevents it. Most tokens have protections against it already;
+                // this is just an extra layer of security.
+                revert WrapAmountTooSmall(params.wrappedToken);
+            }
```

Comment:
We should probably put this inside a function `_ensureValidWrapAmount` (analogous to the one we use for swaps).